File: fem/F/openmp-target/auto.opt/src/mat_con1.f
!C
!C***
!C*** MAT_CON1
!C***
!C
      subroutine MAT_CON1
      use pfem_util
      implicit none

      integer(kind=kint) :: i,k,kk

      allocate (indexLU(0:NP))
      indexLU= 0

      do i= 1, NP
        indexLU(i)= indexLU(i-1) + INLU(i)
      enddo

      NPLU= indexLU(NP)

      allocate (itemLU(NPLU))

      do i= 1, NP
        do k= 1, INLU(i)
           kk = k + indexLU(i-1)
          itemLU(kk)=      IALU(i,k)
        enddo
      enddo

      deallocate (INLU, IALU)

      end subroutine MAT_CON1

File: fem/C++/kokkos/team/cpu-chunk-sweep/src/mat_con1.cpp
/**
 ** MAT_CON1
 **/
#include <stdio.h>
#include "pfem_util.h"
#include "allocate.h"
extern FILE* fp_log;
void MAT_CON1()
{
  int i,k,kk;
  
  indexLU_g=View1D<KINT>("indexLU",N+1);
  auto indexLU_h=Kokkos::create_mirror_view(Kokkos::HostSpace{}, indexLU_g); //host(mirror)
  for(i=0;i<N+1;i++) indexLU_h(i)=0;
  
  for(i=0;i<N;i++){
    indexLU_h(i+1)=indexLU_h(i)+INLU[i];
  }
  
  NPLU=indexLU_h(N);
  
  itemLU_g=View1D<KINT>("itemLU",NPLU);
  auto itemLU_h=Kokkos::create_mirror_view(Kokkos::HostSpace{}, itemLU_g); //host(mirror)
  
  for(i=0;i<N;i++){
    for(k=0;k<INLU[i];k++){
      kk=k+indexLU_h(i);
      itemLU_h(kk)=IALU[i][k]-1;
    }
  }

  Kokkos::deep_copy(indexLU_g,indexLU_h); //host to device
  Kokkos::deep_copy(itemLU_g,itemLU_h); //host to device
 
  deallocate_vector(INLU);
  deallocate_vector(IALU);
}
